### ISSUE TITLE
Error with texts

### ISSUE THREAD

**Issue by Tim1212** [gdkchan/HDATextTool#5]
Hello, I tryed change some text, and when I'm changing the text with "\ "for ex. I've created text with text like "I've heard [var]\x0001 dog barking and rushed here" and then i extract it and get messed up .txt file, with text like this "barkin\x3D10\x3800\x4843r\x3537oovd\x403Ee[end]"... Is there solution for this?

**Comment by gdkchan** [comment]
Well, when extracting (and creating the texts) you need to supply two files to the tool, the text file and the pointers file. Make sure that you're providing the correct files to the tool, and when creating, that you're inserting new data on the **same files**, and when you're extracting again, that you're extracting from the exact same two files too. IIRC this kind of problem happens when you use the wrong files at some point of the process.

**Comment by Tim1212** [comment]
No, you didn't get me. So both are correct files, but when I'am importing text with type like this ""I've heard [var]\x0001 dog barking and rushed here", I create txt, and then after i use again -xtxt to see the .txt file and it messes up like this "barkin\x3D10\x3800\x4843r\x3537oovd\x403Ee[end]"

**Comment by Tim1212** [comment]
I tryed already same

**Comment by gdkchan** [comment]
Can you say which file exactly you're trying to edit? Ill test it here.

**Comment by Tim1212** [comment]
File_00011.bin File_00012.bin

**Comment by gdkchan** [comment]
and the hda file...?

**Comment by gdkchan** [comment]
Can't reproduce. I extracted the text file, edited, inserted back, extracted again and everything is OK.
Commands used:

```
hdatexttool -xhda EVTMSG00.HDA EVTMSG00
hdatexttool -xtxt File_00011.bin File_00012.bin texts.txt
hdatexttool -ctxt texts.txt File_00011.bin File_00012.bin
hdatexttool -xtxt File_00011.bin File_00012.bin texts.txt
```

File edited with notepad. Make sure you're passing the arguments in the correct order.

**Comment by Tim1212** [comment]
Ok, thank you.

**Comment by gdkchan** [comment]
Was the issue resolved?

**Comment by Tim1212** [comment]
No, still the same :(

**Comment by Tim1212** [comment]
It looks like this when i export it...
![error](https://cloud.githubusercontent.com/assets/25753888/22944600/7c03bbf2-f2f1-11e6-83cd-e9e360a58a4c.png)

**Comment by Tim1212** [comment]
Issue resolved, After "I've heard" must set this " , " and then text works
![fixed](https://cloud.githubusercontent.com/assets/25753888/22978400/db6347c4-f392-11e6-9c15-ed7d634f47f9.png)

**Comment by gdkchan** [comment]
Sorry, I couldn't reproduce the issue yesterday because it actually happens when you keep the original text. Turns out the encoder had a very case specific bug that caused this issue. Should be fixed now on lastest commit (version 0.3.1), please retest.

**Comment by Tim1212** [comment]
Tested, works. Thanks alot!
